### DISCUSSION TITLE
Stop conversion to single lines for short control flow blocks

### DISCUSSION THREAD

**Discussion by spaceunifyfifty** [scalameta/scalafmt#4165] [Q&A]
for short blocks, scalafmt is auto-formatting
```scala
if x then
  println("y")
```
as
```scala
if x then println("y")
```
and likewise
```scala
for i <- 1 to 10 do
  println("y")
```
becomes
 ```scala
for i <- 1 to 10 do println("y")
```

How can I disable this opinionated style correction while still keeping other basic scalafmt features like indentation-size correction? I do not want to disable scalafmt entirely, but I will have to if this is not possible.

**Comment by tgodzik** [comment]
Does `newlines.source = keep` setting help in you case?

**Reply by spaceunifyfifty** [reply]
No, this did not work. This is why a preset with all rules disabled is a good idea; so developers don't need to trial-and-error all rules that sound like they kinda maybe possibly might be responsible for the unwanted corrections.

**Reply by kitbellew** [reply]
1. when you say "it didn't work", can you show what was produced (and what you configured)?
2. the tool is not a collection of configurable rules (although there are some, like rewrites, which can be enabled); for the most part, it will always apply some formatting changes but they can be slightly configured in how exactly they will behave (but there's no way to turn it off).

what you are asking for is currently impossible; each "rule" sets its own indentation policies, depending on where _it_ will impose a break, and if there is no rule, then there's no indentation, since we don't know where _you_ have imposed breaks.

however, `newlines.source = keep` is a work-in-progress attempt to preserve some (where possible) breaks that the user has introduced. if "it doesn't work", it is either because this case has not been implemented yet (which is why you could show what you did), or because it contradicts some other assumptions and newline cannot be preserved.

**Reply by spaceunifyfifty** [reply]
Here is my `.scalafmt`
```
version = "3.7.15"
runner.dialect = scala3
indent.main = 2
indent.callSite = 2
newlines.source = keep 
```

What is produced is that 
```scala
if true then 
  println("ok")
```
turns into
```scala
if true then println("ok")
```
